core/cache/charmconfigwatcher.go

```diff
@@ -59,6 +59,7 @@ type CharmConfigWatcher struct {
 	charmURL   string
 	branchName string
 
+	metrics        *ControllerGauges
```

Comment:
I would use the specific gauges; perhaps even the actual `Inc` functions as members, in order to limit the knowledge that this type requires.